core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -224,6 +224,19 @@ private[spark] object JsonProtocol {
     ("Spark Version" -> SPARK_VERSION)
   }
 
+  def executorMetricsUpdateToJson(metricsUpdate: SparkListenerExecutorMetricsUpdate): JValue = {
+    val execId = metricsUpdate.execId
+    val taskMetrics = metricsUpdate.taskMetrics
+    ("Event" -> Utils.getFormattedClassName(metricsUpdate)) ~
+    ("Executor ID" -> execId) ~
+    ("Metrics Updated" -> taskMetrics.map(s =>
+      ("Task ID" -> s._1) ~
+        ("Stage ID" -> s._2) ~
+        ("Stage Attempt ID" -> s._3) ~
+        ("Task Metrics" -> taskMetricsToJson(s._4)))
```

Comment:
instead of underscores, could you rewrite this as:
```
("Metrics Updated" -> taskMetrics.map { case (taskId, stageId, stageAttemptId, metrics)
  ("Task ID" -> taskId) ~
  ("Stage ID" -> stageId) ~
  ("Stage Attempt ID" -> stageAttemptId) ~
  ("Task Metrics" -> taskMetricsToJson(metrics))
)
```